x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/MlNativeDataFrameAnalyticsIntegTestCase.java

```diff
@@ -119,20 +120,21 @@ protected void waitUntilAnalyticsIsStopped(String id) throws Exception {
     }
 
     protected void waitUntilAnalyticsIsStopped(String id, TimeValue waitTime) throws Exception {
-        assertBusy(() -> assertThat(getAnalyticsStats(id).get(0).getState(), equalTo(DataFrameAnalyticsState.STOPPED)),
-                waitTime.getMillis(), TimeUnit.MILLISECONDS);
+        assertBusy(() -> assertIsStopped(id), waitTime.getMillis(), TimeUnit.MILLISECONDS);
     }
 
     protected List<DataFrameAnalyticsConfig> getAnalytics(String id) {
         GetDataFrameAnalyticsAction.Request request = new GetDataFrameAnalyticsAction.Request(id);
         return client().execute(GetDataFrameAnalyticsAction.INSTANCE, request).actionGet().getResources().results();
     }
 
-    protected List<GetDataFrameAnalyticsStatsAction.Response.Stats> getAnalyticsStats(String id) {
+    protected GetDataFrameAnalyticsStatsAction.Response.Stats getAnalyticsStats(String id) {
         GetDataFrameAnalyticsStatsAction.Request request = new GetDataFrameAnalyticsStatsAction.Request(id);
         GetDataFrameAnalyticsStatsAction.Response response = client().execute(GetDataFrameAnalyticsStatsAction.INSTANCE, request)
             .actionGet();
-        return response.getResponse().results();
+        List<GetDataFrameAnalyticsStatsAction.Response.Stats> stats = response.getResponse().results();
+        assertThat(stats.size(), equalTo(1));
```

Comment:
If the size is greater than 1 it might be worth printing the whole list in the message.  It will be very annoying if this ever fails and all we find out is "expected 1, got 2".

Reply:
Done.